src/types/dmabuf.rs

```diff
@@ -0,0 +1,27 @@
+//! Support for the DMABuf type
+
+use wayland_sys::server::wl_display as wl_server_display;
+use wlroots_sys::{wl_display, wlr_renderer, wlr_linux_dmabuf_v1, 
+    wlr_linux_dmabuf_v1_create, wlr_linux_dmabuf_v1_destroy};
+
+#[derive(Debug)]
+pub struct Dmabuf {
+    dmabuf: *mut wlr_linux_dmabuf_v1
+}
+
+impl Dmabuf {
+    pub(crate) unsafe fn new(display: *mut wl_server_display, renderer: *mut wlr_renderer) -> Option<Self> {
```

Comment:
It's not actually possible to create this because it's `pub(crate) unsafe`. You should be exposing a safe way to create a `Dmabuf`. Since it takes a `display` and `renderer` you should pass in a `&Compositor` (which owns the `wl_display` pointer and the `GenericRenderer` which has the `wlr_renderer` pointer). You'll need to also ensure that the `GenericRenderer` is present since it could be built without it. If it's not there, just return `None`.

Reply:
So to get the compositor without running into uses, should I pass in a Handle to Compositor or no?

Reply:
Pass it a &Compositor, not a CompositorHandle. If you pass a handle you must upgrade it and then deal with it failing. If you force the caller to make that decision it makes this api much easier to use (because you don't need to return a result and possibly fail).